### PR TITLE
Add api for updating searcher with selected component

### PR DESCRIPTION
**Pull request by MichaelMauderer**
### Pull Request Description

Implements [#182593882](https://www.pivotaltracker.com/story/show/182633582).

Adds API for updating Searcher with the selected suggestion. To-dos are left in the places where this will be implemented in a subsequent PR for the next task.

**Example of the debug output generated by typing into the searcher.**

ShowSearcherInputLog

**Example of the debug output generated by selecting something in the searcher.**

ShowSearcherSelectionLog

### Important Notes
 
For testing/QA set the `app/gui/src/lib.rs:145` logging initialization to `DEBUG` to see the console output. 

[ci no changelog needed]

### Checklist

Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed: Enso GUI was tested when built using BOTH
        `./run ide build` and `./run ide watch`.

### CODE REVIEW

File: app/gui/src/controller/searcher.rs [hunk 1]
```diff
@@ -696,6 +696,18 @@ impl Searcher {
         }
     }
 
+    /// Preview the suggestion in the searcher.
+    pub fn preview_entry_as_suggestion(&self, index: usize) {
+        tracing::debug!("Previewing entry: {:?}", index);
```
Comment (farmaazon) [inline app/gui/src/controller/searcher.rs:701]: The Acceptance criteria for this task say:
> there is log message from the Searcher with the received expression content, and required imports.

So, technically, this should print an expected expression, not just an index.

If you already have this implemented on the other branch, we can chat if it's plausible to move some code here, or just skip it.

**Review by farmaazon — CHANGES_REQUESTED** [review]

File: app/gui/src/controller/searcher.rs (hunk 1 above)
Comment (MichaelMauderer) [reply app/gui/src/controller/searcher.rs:701]: You are right. This is what happens in the `preview_suggestion`, which is used directly by the new component browser, but not here in the method used by the old searcher.

In my current branch, this is already unified to a common method for both. I can backport this to be used here to get a nicer log output.

**Review by wdanilo — COMMENTED** [review]
The way you've presented these screenshots is wonderful, thanks for that!

**Review by wdanilo — APPROVED** [review]

File: app/gui/src/controller/searcher.rs (hunk 1 above)
Comment (farmaazon) [reply app/gui/src/controller/searcher.rs:701]: Well, if it is an old searcher only, then I think there is no need to do that.

**Review by farmaazon — APPROVED** [review]

**Pull request merged**